src/DotNetEnv/Parsers.cs

```diff
@@ -126,8 +126,24 @@ from value in Octal.Repeat(1, 3).Text()
             select ToUtf8Char(value)).Try();
 
         internal static readonly TextParser<string> Utf8Char =
-            (from value in HexByte.Repeat(1, 4)
-            select ToUtf8Char(value)).Try();
+            (from firstByte in HexByte
+                from nextBytes in HexByte.Repeat(GetUtf8CharByteCount(firstByte) - 1)
+                select ToUtf8Char(new[] { firstByte }.Concat(nextBytes))).Try();
+
+        /// <summary>
+        /// Returns byte-count of a UTF-8 character by its first byte.
+        /// </summary>
+        /// <param name="firstByte">The first byte of the UTF-8 char</param>
+        /// <returns>the byte-count of a UTF-8 char.</returns>
+        /// <remarks>https://en.wikipedia.org/wiki/UTF-8#Description</remarks>
+        private static int GetUtf8CharByteCount(byte firstByte)
+            => firstByte switch
```

Comment:
Hmm, I suppose using newer language features in the library only means compiling the library requires a newer compiler (I also realize lang 9 is from 2020, so not that new), and that client applications could still compile on an older compiler if needed, right? Please sanity check me on that. I don't want this feature to force a bunch of projects to have to upgrade to use this library.

Reply:
How different would this be if you used a less fancy switch formulation? Could you just write a function that was a series of ifs in this context?

Reply:
I thought that LanguageVersion is just relevant for compile-time, which wouldn't affect any client applications.
But it seems that it may cause runtime errors too, that would be ugly I guess.

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-versioning

==> I will change that

Reply:
I pushed a ternary-expression which is more concise imho compared to if-statements.
But if you like the if statements better I will change it, something like that:
```
if (firstByte < (byte)'\x80')
    return 1;
if (firstByte < (byte)'\xE0')
    return 2;
if (firstByte < (byte)'\xF0')
    return 3;
return 4;
```